gdc_ng_models/models/cohort.py

```diff
@@ -0,0 +1,222 @@
+"""Data model to describe cohorts.
+
+A cohort defines a set of cases that are of interest to a user. The concept
+of a cohort is central to the user experience in the next-gen data portal. The
+cohort data model includes the following entities:
+    AnonymousContext: Used to authorize changes to a cohort
+    Cohort: Defines the basic properties (name, id, context) of a cohort
+    CohortFilter: Defines the filter used to generate a cohort case set
+    CohortSnapshot: Defines the set of cases for a static cohort
+"""
+
+import uuid
+
+from sqlalchemy import orm, ARRAY, BigInteger, Boolean, Column, ForeignKey, Text
+from sqlalchemy.ext import declarative
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+from gdc_ng_models.models import audit
+
+Base = declarative.declarative_base()
+
+
+class AnonymousContext(Base, audit.AuditColumnsMixin):
+    """An anonymous context to drive authorization for cohort manipulation.
+
+    Provides authorization capabilities in lieu of a login (something the GDC
+    data portal does not and cannot implement). When a cohort is created, it
+    is associated with a context ID that establishes identity similar to a
+    bearer token. When a request is made to manipulate a cohort, the context ID
+    must be provided as part of that request to authorize a change.
+
+    Attributes:
+        id: A UUID identifier for the context.
```

Comment:
I'm not sure if the 'types' are needed here?

Reply:
Agreed, these are unnecessary. I've removed the type references in the latest commit.